utilities/transactions/pessimistic_transaction.h

```diff
@@ -236,8 +418,100 @@ class WriteCommittedTxn : public PessimisticTransaction {
   Status CommitInternal() override;
 
   Status RollbackInternal() override;
+
+  // Column families that enable timestamps and whose data are written when
+  // indexing_enabled_ is false. If a key is written when indexing_enabled_ is
+  // true, then the corresponding column family is not added to cfs_with_ts
+  // even if it enables timestamp.
+  std::unordered_set<uint32_t> cfs_with_ts_tracked_when_indexing_disabled_;
 };
 
+template <typename TValue>
```

Comment:
Very minor but since these are private templates only used in `WriteCommittedTxn`, their implementation could be moved to the .cc file. (Also, the same goes for the methods that use these templates like `Put` / `Delete` etc.)